### DISCUSSION TITLE
What size is the location's x, y, and z based on?

### DISCUSSION THREAD

**Discussion by EthanKyleKim** [br-g/openf1#95] [Q&A] [answered]
Hello everyone I'm developing to show race information using data from OpenF1.

I want to make the circuit 2D and express the position of each driver on top of it.

But I'm not sure what criteria to generate the map with x, y, z in the location.

Does anyone have any ideas on what criteria there are or how to draw a map?

Thank you for reading it.

**Comment by adityakotha03** [comment, marked as answer]
The x, y and z are not fixed to any one point in my testing. They vary every race so it's hard to make one and copy it for others. The only way is to create a custom one for every track, or you could just find the new origin for each track and shift the x, y and z values for each driver based on that.

**Reply by EthanKyleKim** [reply]
Do you know where to get the x, y, and z information provided by openf1?

**Reply by adityakotha03** [reply]
Location end point from the API.

**Reply by mohamedhadrami** [reply]
If you hit the endpoint, 

https://api.multiviewer.app/api/v1/circuits/${circuitKey}/${year}

you can get the actual x and y for the track itself, separate from the driver's location data. I've tried it before and it seems to line up driver location data. You can even see the racing line!

**Reply by adityakotha03** [reply]
> If you hit the endpoint,
> 
> [https://api.multiviewer.app/api/v1/circuits/${circuitKey}/${year}](https://api.multiviewer.app/api/v1/circuits/$%7BcircuitKey%7D/$%7Byear%7D)
> 
> you can get the actual x and y for the track itself, separate from the driver's location data. I've tried it before and it seems to line up driver location data. You can even see the racing line!

Hey is there any documentation for the API support on multiviewer?? Any idea on how they support this stuff or where to find other stuff in their API?
Thanks.

**Reply by mohamedhadrami** [reply]
> > If you hit the endpoint,
> > [https://api.multiviewer.app/api/v1/circuits/${circuitKey}/${year}](https://api.multiviewer.app/api/v1/circuits/$%7BcircuitKey%7D/$%7Byear%7D)
> > you can get the actual x and y for the track itself, separate from the driver's location data. I've tried it before and it seems to line up driver location data. You can even see the racing line!
> 
> Hey is there any documentation for the API support on multiviewer?? Any idea on how they support this stuff or where to find other stuff in their API? Thanks.

This is just what I found from looking at other people's code in the community. If you want more endpoint, my guess would be to download the multiviewer app and just monitor the network traffic and see where its trying to reach. I haven't done much diving into this yet.
